src/config/job-config/jobconfig.service.ts

```diff
@@ -0,0 +1,132 @@
+import { Inject, Injectable } from "@nestjs/common";
+import {
+  CREATE_JOB_ACTION_FACTORIES,
+  JobActionFactory,
+  JobConfig,
+  JobConfigListOptions,
+  JobConfigOptions,
+  JobDto,
+  JobOperation,
+  JobOperationOptions,
+  UPDATE_JOB_ACTION_FACTORIES,
+} from "./jobconfig.interface";
+import Ajv from "ajv";
+import { JobConfigSchema } from "./jobconfig.schema";
+import { load } from "js-yaml";
+import * as fs from "fs";
+import { CreateJobDto } from "../../jobs/dto/create-job.dto";
+import { StatusUpdateJobDto } from "../../jobs/dto/status-update-job.dto";
+import { ConfigService } from "@nestjs/config";
+
+/**
+ * Service representing a parsed jobconfig.yaml file.
+ */
+@Injectable()
+export class JobConfigService {
+  private readonly jobs: Record<string, JobConfig>;
+  private readonly filePath: string;
+
+  constructor(
+    @Inject(CREATE_JOB_ACTION_FACTORIES)
+    private create_factories: Record<string, JobActionFactory<CreateJobDto>>,
+    @Inject(UPDATE_JOB_ACTION_FACTORIES)
+    private update_factories: Record<
+      string,
+      JobActionFactory<StatusUpdateJobDto>
+    >,
+    configService: ConfigService,
+  ) {
+    this.filePath = configService.get<string>("jobConfigurationFile") || "";
+    this.jobs = this.loadJobConfig(this.filePath);
+  }
+
+  public get(jobType: string) {
+    return this.jobs[jobType]; // TODO error handling
+  }
+
+  public get allJobConfigs(): Readonly<Record<string, JobConfig>> {
+    return this.jobs;
+  }
+
+  /**
+   * Load jobconfig.yaml (or json) file.
+   * Expects one or more JobConfig configurations (see JobConfig.parse)
+   * @param filePath path to json config file
+   * @returns
+   */
+  private loadJobConfig(filePath: string): Record<string, JobConfig> {
+    // If no file is given don't configure any jobs
+    if (!filePath) {
+      return {};
+    }
+    const yaml = fs.readFileSync(filePath, "utf8");
+    const jobConfigListOptions = load(yaml, { filename: filePath });
+
+    // TODO: Do we like the schema-based validation, or should it be pure typescript?
```

Comment:
I'm not sure validation is even necessary, since all the constructors throw errors if the yaml doesn't follow the expected structure. I expect ajv gives clearer error messages though.